Vowpal Wabbit - Contextual Bandits.ipynb

```diff
@@ -0,0 +1,705 @@
+     "text": [
+      "root\n",
+      " |-- input: string (nullable = true)\n",
+      " |-- json: struct (nullable = true)\n",
+      " |    |-- EventId: string (nullable = true)\n",
+      " |    |-- _label_probability: float (nullable = true)\n",
+      " |    |-- _labelIndex: integer (nullable = true)\n",
+      " |    |-- _label_cost: float (nullable = true)\n",
+      " |-- EventId: string (nullable = true)\n",
+      " |-- rewards: struct (nullable = false)\n",
+      " |    |-- reward: float (nullable = true)\n",
+      " |-- probLog: float (nullable = true)\n",
+      " |-- chosenActionIndex: integer (nullable = true)\n",
+      " |-- splitId: integer (nullable = false)\n",
+      "\n",
+      "DataFrame[json: struct<EventId:string,_label_probability:float,_labelIndex:int,_label_cost:float>, EventId: string, rewards: struct<reward:float>, probLog: float, chosenActionIndex: int, splitId: int]\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      "[Stage 17:>                                                         (0 + 1) / 1]\r"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+------------------------------------------------------+--------------------------------+-------+---------+-----------------+-------+\n",
+      "|json                                                  |EventId                         |rewards|probLog  |chosenActionIndex|splitId|\n",
+      "+------------------------------------------------------+--------------------------------+-------+---------+-----------------+-------+\n",
+      "|{fbe7a11d120b4df4bf23b836de8a29d1, 0.8166667, 9, 0.0} |fbe7a11d120b4df4bf23b836de8a29d1|{0.0}  |0.8166667|9                |0      |\n",
+      "|{0074434d3a3a46529f65de8a59631939, 0.8166667, 9, -1.0}|0074434d3a3a46529f65de8a59631939|{-1.0} |0.8166667|9                |0      |\n",
+      "|{9077f996581148978a0ebe2484260dab, 0.8166667, 9, -1.0}|9077f996581148978a0ebe2484260dab|{-1.0} |0.8166667|9                |0      |\n",
+      "+------------------------------------------------------+--------------------------------+-------+---------+-----------------+-------+\n",
+      "\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      "[Stage 17:==========================================================(1 + 0) / 1]\r",
+      "\r",
+      "                                                                                \r"
+     ]
+    }
+   ],
+   "source": [
+    "from synapse.ml.vw import VowpalWabbitDSJsonTransformer\n",
+    "\n",
+    "df_ready = (\n",
+    "    VowpalWabbitDSJsonTransformer()\n",
+    "    .setDsJsonColumn(\"input\")\n",
+    "    .transform(df)\n",
+    "    .withColumn(\"splitId\", F.lit(0))\n",
+    "    .repartition(2)\n",
+    ")\n",
+    "df_ready.printSchema()\n",
+    "\n",
+    "# exclude JSON as it's too messy\n",
```

Comment:
nit: put the drop on df_ready. Also maybe call this dataframe somethinmore natural like "train"